src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+## Custom Primary Keys
+
+By default, all DataStore models have an `id` field by default that is automatically populated on the client with a UUID v4, allowing DataStore to generate non-colliding globally unique identifiers in a scalable way.
+
+While UUIDs have desirable properties (they are large, non-sequential and opaque), there are times when a custom primary key is needed. For instance, to:
+
+
+* Have friendly/readable identifiers (surrogate/opaque vs. natural keys)
+* Define composite primary keys
+* Customize data partitioning to optimize for scale (especially important when planning to handle large amounts of data in short periods of time)
+* Selectively synchronize data to clients (e.g. by fields like `deviceId`, `userId` or similar)
+* Prioritize the sort order in which objects are returned by the sync queries
+* Make existing data consumable and syncable by DataStore
+
+### Schema
+
+A schema with the typical `id` field looks like this:
+
+```graphql
+type Todo4 @model {
+	id: ID!
+	name: String!
+	description: String
+}
+```
+
+With custom primary keys, you can provide a unique value using the `@primaryKey` directive:
+
+```graphql
+type Todo @model {
```

Comment:
In the rest of our docs, we typically use "Todo" (or "Post" / Comment"). These are also the out-of-the-box schemas that the CLI generates, and I do find value in keeping them the same for a few reasons. 1) If I'm experimenting with new DS concepts using the happy path, I can easily copy / paste into an existing project (people usually default to Todo's) without having to change any client code, 2) it reduces the cognitive overload (however small) regarding thinking about new relational mappings, since I'm already looking at Todo examples everywhere else.
Thoughts on that? I do see value in including more real-world examples, as well.

Reply:
Discuss offline, will update